app/files.js

```diff
@@ -519,6 +618,60 @@ const roomCreate = {
   }
 }
 
+const shareRoom = {
+  key: "shareRoom",
+  noun: "Room",
+  display: {
+    label: "Share Room",
+    description: "Share a room with a user."
+  },
+  operation: {
+    inputFields: [
+      {
+        label: "Room",
+        key: "roomId",
+        required: true,
+        altersDynamicFields: true,
+        dynamic: "roomCreated.id.title"
+      },
+      {
+        label: "User",
+        key: "userId",
+        required: true,
+        dynamic: "userAdded.id.displayName"
+      },
+      {
+        label: "Role",
+        key: "access",
+        required: true,
+        dynamic: "shareRoles.id.name"
+      }
+    ],
+    /**
+     * @param {ZObject} z
+     * @param {Bundle<SessionAuthenticationData, ShareOptions>} bundle
+     * @returns {Promise<ShareData>}
+     */
+    async perform(z, bundle) {
+      const client = new Client(bundle.authData.baseUrl, z.request)
+      const files = new FilesService(client)
+      /** @type {ShareBody} */
+      const body = {
+        invitations: [{
+          access: bundle.inputData.access,
+          id: bundle.inputData.userId
+        }],
+        notify: true, // Default value from DocSpace
+        message: "Invitation message" // Default value from DocSpace
```

Comment:
> Usually, the client doesn't need to send default values to the server since the second one will substitute them if needed. So, why do we send them?

This is the default message sent when a user is invited to DocSpace. For clarity, I've replaced the text with "Invitation from Zapier"